sympy/polys/matrices/domainmatrix.py

```diff
@@ -100,62 +150,100 @@ def from_Matrix(cls, M, **kwargs):
 
         Parameters
         ==========
 
-        K: Domain. The domain to convert the elements to.
+        K : Represents the desired domain or field
+
+        Returns
+        =======
+
+        DomainMatrix
+            DomainMatrix with the desired domain or field
 
         Examples
-        ========
+        =====
 
+        >>> from sympy import ZZ, ZZ_I
         >>> from sympy.polys.matrices import DomainMatrix
-        >>> from sympy import QQ, ZZ
-        >>> A = DomainMatrix([[ZZ(1), ZZ(3)]], (1, 2), ZZ)
-        >>> A.convert_to(QQ)
-        DomainMatrix([[1, 3]], (1, 2), QQ)
+        >>> domainmatrix = DomainMatrix([
+        ...    [ZZ(1), ZZ(2)],
+        ...    [ZZ(3), ZZ(4)]], (2, 2), ZZ)
 
-        See Also
-        ========
-
-        DomainMatrix.unify, DomainMatrix.convert_to
+        >>> domainmatrix.convert_to(ZZ_I)
+        DomainMatrix([[1, 2], [3, 4]], (2, 2), ZZ_I)
 
         """
         return self.from_rep(self.rep.convert_to(K))
 
     def to_field(self):
+        r"""
+        Returns a DomainMatrix with the appropriate field
+
+        Returns
+        =======
+
+        DomainMatrix
+            DomainMatrix with the appropriate field
+
+        Examples
+        ========
+
+        >>> from sympy import ZZ
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> domainmatrix = DomainMatrix([
+        ...    [ZZ(1), ZZ(2)],
+        ...    [ZZ(3), ZZ(4)]], (2, 2), ZZ)
+
+        >>> domainmatrix.to_field()
+        DomainMatrix([[1, 2], [3, 4]], (2, 2), QQ)
+
+        """
         K = self.domain.get_field()
         return self.convert_to(K)
 
     def unify(self, other):
-        """Converts self and other to a common domain
+        r"""
+        Unify the domains of self and other
 
         Parameters
         ==========
 
-        other: other DomainMatrix to be unified
+        other : another DomainMatrix
 
         Returns
         =======
 
-        Two Matrices that contain the elements of the input
-        matrices but they now belong to a unified Domain
+        (dM1, dM2)
+            dM1, dM2 DomainMatrix matrices with unified Domain
 
         Examples
         ========
 
-        >>> from sympy.polys.matrices import DomainMatrix
         >>> from sympy import ZZ, QQ
-        >>> A = DomainMatrix([[ZZ(1), ZZ(2), ZZ(3)]], (1, 3), ZZ)
-        >>> B = DomainMatrix([[QQ(1, 2), QQ(3, 5)]], (1, 2), QQ)
-        >>> A.unify(B)
-        (DomainMatrix([[1, 2, 3]], (1, 3), QQ), DomainMatrix([[1/2, 3/5]], (1, 2), QQ))
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> domainmatrix1 = DomainMatrix([
+        ...    [ZZ(1), ZZ(2)],
+        ...    [ZZ(3), ZZ(4)]], (2, 2), ZZ)
+        >>> domainmatrix2 = domainmatrix1.convert_to(QQ)
+
+        >>> dM1_new, dM2_new = domainmatrix1.unify(domainmatrix2)
+        >>> dM1_new
+        DomainMatrix([[1, 2], [3, 4]], (2, 2), QQ)
+        >>> dM2_new
+        DomainMatrix([[1, 2], [3, 4]], (2, 2), QQ)
```

Comment:
@hyadav2k I feel like you could let my example remain. It seems to be written in a more simpler way.

Reply:
Sorry, will use your example here, it looks better.